src/mlpack/tests/loss_functions_test.cpp

```diff
@@ -510,4 +511,103 @@ BOOST_AUTO_TEST_CASE(LogCoshLossTest)
   BOOST_REQUIRE_EQUAL(output.n_rows, input.n_rows);
   BOOST_REQUIRE_EQUAL(output.n_cols, input.n_cols);
 }
+
+/**
+ * Simple test for the Cosine Embedding loss function.
+ */
+BOOST_AUTO_TEST_CASE(CosineEmbeddingLossTest)
+{
+  arma::mat input1, input2, y, output;
+  double loss;
+  CosineEmbeddingLoss<> module;
+
+  // Test the Forward function. Loss should be 0 if input1 = input2 and y = 1.
+  input1 = arma::mat(1, 10);
+  input2 = arma::mat(1, 10);
+  input1.ones();
+  input2.ones();
+  y = arma::mat(1, 1);
+  y.ones();
+  loss = module.Forward(input1, input1, y);
```

Comment:
do you mean to be using input1, input2 here, or is that intentional?

Reply:
Hi, This was intentional to show that loss would be zero for same inputs.